src/Database/PostgreSQL/PQTypes/Internal/C/Interface.hs

```diff
@@ -88,6 +91,34 @@ foreign import ccall unsafe "PQclear"
 
 ----------------------------------------
 
+foreign import ccall unsafe "PQgetCancel"
+  c_PQgetCancel :: Ptr PGconn -> IO (Ptr PGcancel)
+
+foreign import ccall unsafe "PQfreeCancel"
+  c_PQfreeCancel :: Ptr PGcancel -> IO ()
+
+foreign import ccall unsafe "PQcancel"
+  c_rawPQcancel :: Ptr PGcancel -> CString -> CInt -> IO CInt
+
+-- | Attempt to cancel currently running query. If the request is successfully
+-- dispatched Nothing is returned, otherwise a textual explanation of what
+-- happened.
+c_PQcancel :: Ptr PGconn -> IO (Maybe String)
+c_PQcancel conn = E.mask $ \restore -> do
+  cancel <- c_PQgetCancel conn
```

Comment:
Can be also written with `bracket` I guess.

Reply:
Right, somewhat I missed that. Will push a fix.